packages/jest-snapshot/src/inline_snapshots.ts

```diff
@@ -264,6 +282,8 @@ const createFormattingParser = (
         return;
       }
 
+      snapshotMatcherNames.shift();
```

Comment:
is it safe to mutate this array in here? Should we make a copy at the top of `createFormattingParser` and mutate that instead?

What I'm afraid of is what happens if I have multiple custom inline matchers and multiple "normal" inline matchers in the same test?

Reply:
I've thought something similar when implementing this, I thought that there must be some tests already cover it up? I'll look into it to make sure.

Reply:
I think creating a copy and mutating that is better than mutating an argument passed in anyways. But it would be good to have a test for multiple custom and multiple normal inline snapshots not stepping on each other regardless if we don't have it 🙂

Reply:
Just to be clear, but we're still mutating the array in `createInsertionParser`. Is that OK? I couldn't think of a better way without mutating and refactoring a lot here.

Reply:
Could you add a test that uses multiple custom inline snapshot matchers?

Reply:
Just did in https://github.com/facebook/jest/pull/9278/commits/d530b1c7e29af3e3f90cfb21f7e4a44cedb9fb1b. Is it what you meant?